src/coders/sequencer-batch.ts

```diff
@@ -118,9 +118,24 @@ export const decodeAppendSequencerBatch = (
   }
 
   return {
-    shouldStartAtBatch: parseInt(shouldStartAtBatch, 16),
+    shouldStartAtElement: parseInt(shouldStartAtElement, 16),
     totalElementsToAppend: parseInt(totalElementsToAppend, 16),
     contexts,
     transactions,
   }
 }
+
+module.exports.sequencerBatch = {
+  encode: (b) => {
+    return (
+      ethers.utils.id(APPEND_SEQUENCER_BATCH_METHOD_ID).slice(0, 10) +
```

Comment:
We are going to need some test coverage around this as I think `encodeAppendSequencerBatch` is going to return a string with a `0x` prefix. Example calldata can be found on transactions here: https://etherscan.io/address/0xed2701f7135eab0D7ca02e6Ab634AD6CbE159Ffb

Look for transactions that call `appendSequencerBatch` and then the calldata should decode and then encode to the same value.

Also what do you think about defining the functions above, like:

```js
const appendSequencerBatch = {
  encode: () => { logic goes here },
  decode: () => { logic goes here }.
}

module.exports = {
  appendSequencerBatch,
  ... other things
}
```

Reply:
Your approach of decoupling the functions from the export looks better than my approach 😅. I'll update the functions that way.

As for the tests around these `encodeAppendSequencerBatch` and `decodeAppendSequencerBatch`, I'll make sure to add these too.

Reply:
2 questions: 

1. Should these new tests be different than the `appendSequencerBatch` tests in `batch-encoder.spec.ts`? 
2. Looked through those transactions on etherscan , but I only see `TransactionBatchAppended` and  `SequencerBatchAppended` being emitted in the event logs. Am I supposed to use the calldata from `SequencerBatchAppended`?